src/test/java/fi/otavanopisto/kuntaapi/test/ResourceMocker.java

```diff
@@ -0,0 +1,194 @@
+    }
+  }
+  
+  public void stop() {
+    started = false;
+    
+    for (I id : subMockers.keySet()) {
+      for (ResourceMocker<?, ?> subMocker : subMockers.get(id)) {
+        subMocker.stop();
+      }
+    }
+    
+    for (MockedResource<R> resource : resources.values()) {
+      for (MappingBuilder mapping : resource.getMappings()) {
+        removeStub(mapping);
+      }
+    }
+    
+    for (MappingBuilder list : statusLists.values()) {
+      removeStub(list);
+    }
+  }
+  
+  public void add(I id, R resource, UrlPattern urlPattern) {
+    MappingBuilder mappingBuilder = get(urlPattern);
+    
+    MappingBuilder okMapping = mappingBuilder
+      .willReturn(aResponse()
+      .withHeader(CONTENT_TYPE, APPLICATION_JSON)
+      .withBody(toJSON(resource)));
+    
+    MappingBuilder notFoundMapping = get(urlPattern)
+      .willReturn(aResponse()
+      .withStatus(404));
+    
+    resources.put(id, new MockedResource<>(resource, okMapping, notFoundMapping));
+  }
+  
+  public Collection<MockedResource<R>> getMockedResources() {
+    return resources.values();
+  }
+  
+  public Collection<MockedResource<R>> getMockedResources(MockedResourceStatus status) {
+    Collection<MockedResource<R>> result = new ArrayList<>();
+    
+    for (MockedResource<R> resource : getMockedResources()) {
+      if (resource.getStatus().equals(status)) {
+        result.add(resource);
+      }
+    }
+    
+    return result;
+  }
+  
+  public List<R> getResources(MockedResourceStatus status) {
+    List<R> result = new ArrayList<>();
+    
+    for (MockedResource<R> mockedResource : getMockedResources(status)) {
+      result.add(mockedResource.getResource());
+    }
+    
+    return result;
+  }
+  
+  public boolean isMocked(I id) {
+    return resources.containsKey(id);
+  }  
+  
+  public void setStatus(I id, MockedResourceStatus status) {
+    MockedResource<R> resource = resources.get(id);
+    if (resource.getStatus() == status) {
+      return;
+    }
+    
+    if (started) {
+      if (resource.getCurrentMapping() != null) {
+        removeStub(resource.getCurrentMapping());
+      }
+    }
+        
+    resource.setStatus(status);
+    
+    if (started) {
+      stubFor(resource.getCurrentMapping());
+      updateStatusLists();
+      
+      if (subMockers.containsKey(id)) {
+        for (ResourceMocker<?, ?> subMocker : subMockers.get(id)) {
+          if (status == MockedResourceStatus.OK) {
+            subMocker.start();
+          } else {
+            subMocker.stop();
+          }
+        }
+      }
+    }
+  }
+
+  public void addStatusList(MockedResourceStatus status, UrlPattern urlPattern) {
+    statusLists.put(status, get(urlPattern));
+  }
+
+  public void addSubMocker(I id, ResourceMocker<?, ?> subMocker) {
+    if (!subMockers.containsKey(id)) {
+      subMockers.put(id, new ArrayList<>());
+    }
+    
+    subMockers.get(id).add(subMocker);
+  }
+
+  public ResourceMocker<?, ?> getSubMocker(I id, int index) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1452)
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1452)